### PR TITLE
Feature/docker commands

### PR DESCRIPTION
**Pull request by evansde77**
@shudgston Starting on the image build helpers.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -12,3 +12,5 @@ PyChef==0.2.3
 virtualenv
 ./venv/cache/gitconfig-0.0.2.tar.gz
 pluggage
+dockerstache
+docker-py
```
Comment (evansde77) [inline requirements.txt]: Need to remove this one, wasnt useful... (Note to self)

File: src/cirrus/docker.py [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+def _docker_build(path, tag):
+    """
+    execute docker build -t <tag> <path> in a subprocess
+    """
+    command = ['docker', 'build', '-t', tag, path]
+    LOGGER.info("Executing docker build command: {}".format(' '.join(command)))
+    stdout = subprocess.check_output(command)
+    LOGGER.info(stdout)
+    return
```
Comment (shudgston) [inline src/cirrus/docker.py]: Remove

File: src/cirrus/docker.py [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+
+def _docker_push(tag):
+    """
+    execute docker push command as a subprocess
+    """
+    command = ['docker', 'push', tag]
+    LOGGER.info("Executing docker push command: {}".format(' '.join(command)))
+    stdout = subprocess.check_output(command)
+    LOGGER.info(stdout)
+    return
```
Comment (shudgston) [inline src/cirrus/docker.py]: Remove return

**Comment by shudgston** [comment]
```$ git cirrus test
[localhost] local: . ./venv/bin/activate && nosetests -w tests/
..................................E.
======================================================================
ERROR: test upload command, mocking out fabric put
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/seanhudgston/src/git/cirrus/tests/unit/cirrus/release_test.py", line 150, in setUp
    self.mock_put = self.patch_put.start()
  File "/Users/seanhudgston/src/git/cirrus/venv/lib/python2.7/site-packages/mock.py", line 1396, in start
    result = self.__enter__()
  File "/Users/seanhudgston/src/git/cirrus/venv/lib/python2.7/site-packages/mock.py", line 1268, in __enter__
    original, local = self.get_original()
  File "/Users/seanhudgston/src/git/cirrus/venv/lib/python2.7/site-packages/mock.py", line 1242, in get_original
    "%s does not have the attribute %r" % (target, name)
AttributeError: <module 'cirrus.release' from '/Users/seanhudgston/src/git/cirrus/src/cirrus/release.py'> does not have the attribute 'put'

----------------------------------------------------------------------
Ran 36 tests in 4.956s

FAILED (errors=1)

Fatal error: local() encountered an error (return code 1) while executing '. ./venv/bin/activate && nosetests -w tests/'

Aborting.```

File: src/cirrus/docker.py [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+
+    provide support for some basic docker operations so that
+    building images can be standardised as part of a workflow
+    """
+    opts = build_parser()
+    config = load_configuration()
+    if not config.has_section('docker'):
+        msg = (
+            "Unable to find docker section in cirrus.conf"
+            )
```
Comment (shudgston) [inline src/cirrus/docker.py]: This can fit a on single line

**Comment by shudgston** [comment]
:+1:

File: src/cirrus/docker.py (hunk 4 above)
Comment (evansde77) [inline src/cirrus/docker.py]: Was going to expand this with a helpful link to the config options documentation I havent written yet... will add a TODO

**Pull request merged**